client/analysisui/events.js

```diff
@@ -0,0 +1,48 @@
+
+'use strict';
+
+var SuperClass = require('../common/superclass');
+
+function Events() {
```

Comment:
do you want to just use the `EventEmitter`? it's the de-facto event system in js.

https://nodejs.org/api/events.html